### PR TITLE
Migrate UndoTest to kotlin

### PR DESCRIPTION
**Pull request by lukstbit**
## Purpose / Description

Migrates the UndoTest.java file to kotlin following the migration guide. In this class one test is set as ignored and the other test is also ignored due to an exception at the last `assumeThat()` call in the method(happens for both java and kotlin versions).

## Fixes

Closes [this PR](https://github.com/ankidroid/Anki-Android/pull/10223).

## How Has This Been Tested?

Ran the usual tests.

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/test/java/com/ichi2/libanki/UndoTest.kt [hunk 1]
```diff
@@ -13,133 +13,135 @@
...
-        assertEquals(1001, c.getLeft());
-        assertEquals(new Counts(0, 1, 0), col.getSched().counts());
-        assertEquals(QUEUE_TYPE_LRN, c.getQueue());
+        Assert.assertEquals(Counts(1, 0, 0), col.sched.counts())
+        var c = col.sched.card
+        Assert.assertEquals(QUEUE_TYPE_NEW, c!!.queue)
+        col.sched.answerCard(c, Consts.BUTTON_THREE)
+        Assert.assertEquals(1001, c.left)
+        Assert.assertEquals(Counts(0, 1, 0), col.sched.counts())
+        Assert.assertEquals(QUEUE_TYPE_LRN, c.queue)
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/libanki/UndoTest.kt:109]: import `Assert`

File: AnkiDroid/src/test/java/com/ichi2/libanki/UndoTest.kt [hunk 2]
```diff
@@ -13,133 +13,135 @@
...
+        Assert.assertEquals(Counts(2, 0, 0), col.sched.counts())
         // performing a normal op will clear the review queue
-        c = col.getSched().getCard();
-        col.getSched().answerCard(c, Consts.BUTTON_THREE);
-        assertThat(col.undoType(), is(instanceOf(Collection.UndoReview.class)));
-        col.save("foo");
+        c = col.sched.card
+        col.sched.answerCard(c!!, Consts.BUTTON_THREE)
+        @KotlinCleanup("Change `is`() to equalTo()")
+        MatcherAssert.assertThat(col.undoType(), Matchers.`is`(Matchers.instanceOf(UndoReview::class.java)))
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/libanki/UndoTest.kt:140]: import `MatcherAssert` and `Matchers`

File: AnkiDroid/src/test/java/com/ichi2/libanki/UndoTest.kt [hunk 3]
```diff
@@ -13,133 +13,135 @@
...
+        col.reset()
+        Assert.assertEquals(Counts(2, 0, 0), col.sched.counts())
         // performing a normal op will clear the review queue
-        c = col.getSched().getCard();
-        col.getSched().answerCard(c, Consts.BUTTON_THREE);
-        assertThat(col.undoType(), is(instanceOf(Collection.UndoReview.class)));
-        col.save("foo");
+        c = col.sched.card
+        col.sched.answerCard(c!!, Consts.BUTTON_THREE)
+        @KotlinCleanup("Change `is`() to equalTo()")
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/libanki/UndoTest.kt:139]: You can add this at the top of the class in future

**Review by david-allison — CHANGES_REQUESTED** [review]

**Comment by lukstbit** [comment]
All requests for changes have been handled.

**Review by david-allison — APPROVED** [review]

File: AnkiDroid/src/test/java/com/ichi2/libanki/UndoTest.kt (hunk 1 above)
Comment (david-allison) [reply AnkiDroid/src/test/java/com/ichi2/libanki/UndoTest.kt:111]: This wasn't handled, not going to block on it

**Review by mikehardy — APPROVED** [review]
Looks great, thanks

**Pull request merged**